src/objects/ConversationManager.js

```diff
@@ -38,14 +38,56 @@ ConversationManager.prototype.getCurrentText = function () {
   return this.conversation[this.idx]['text'];
 };
 
-ConversationManager.prototype.getResponses = function () {
+ConversationManager.prototype.getResponses = function (game) {
   if (this.conversation === null) {
     return [''];
   }
+  var responses = this.conversation[this.idx]['responses'];
+  var ret = [];
+  for (var i = 0; i < responses.length; i++) {
+    if ('conditions' in responses[i]) {
+      var conditionsNeeded = 0;
+      var conditionsMet = 0;
+      for (var condition in responses[i]['conditions']) {
+        if (checkCondition(game, condition, responses[i]['conditions'][condition])) {
+          conditionsMet++;
+        }
+        conditionsNeeded++;
+      }
+      if (conditionsMet >= conditionsNeeded) {
+        ret.push(responses[i]); //if all conditions are met, display response
+      }
+    } else {
+      ret.push(responses[i]);   //no conditions on this response, display it
+    }
+  }
 
-  return this.conversation[this.idx]['responses'];
+  return ret;
 };
 
+function checkCondition(game, condition, value) {
+  if (condition.startsWith('var')) {
+    var variable = condition.substring(3);
+    if (value.startsWith('!')) {
+      if (!(variable in game.player.variables) || game.player.variables[variable] !== value.substring(1)) {
+        return true;    //player does not have this variable set, or has it set to a different value
+      }
+    } else if (variable in game.player.variables && game.player.variables[variable] === value) {
+      return true;      //player has this variable set to this value
+    }
+  } else if (condition.startsWith('inv')) {
+    var item = condition.substring(3);
+    if (value.startsWith('!')) {
+      if (game.player.inventory.indexOf(item) === -1) {
+        return true;    //player does not have this inventory item
```

Comment:
Should this be true? If the value starts with !, does that mean it's supposed to be the inverse?